cores/esp8266/WString.cpp

```diff
@@ -58,70 +58,71 @@ String::String(StringSumHelper &&rval) {
 #endif
 
 String::String(char c) {
+    uint32_t buf = (uint32_t)c;
```

Comment:
This assumes little-endian. Yeap, sure, there are no such thing as big-endian esp8266; but then someone will copy-paste this code, and kitten will die. It worth at least leaving comment on this dependency (and intent: alignment).
More alternatives - `const char buf __attribute__((__aligned__(4))) = c;` or `union {const char c; uint32_t align; } buf = {c};`, YMMV.